### PR TITLE
[MIG][16.0] Migration of vault_share

### PR DESCRIPTION
**Pull request by fkantelberg**
Based on #611 

The migration needed quite a rework because of the JS framework changes. (I hope the next are easier) I tested every functionality during migration but plan to do another round again soon.

There are some minor fixes which were possible to do:

- Creating shares is now possible using the views
- Better file handling. Inside of an entry you can share now

### CODE REVIEW

**Comment by CarlosRoca13** [comment]
Please rebase to get changes :)

**Review by CarlosRoca13 — APPROVED** [review]
It looks good to me :+1:

**Comment by pedrobaeza** [comment]
@fkantelberg this one doesn't contain commit history. Please preserve it for also analyzing the specific changes done for 16.0

**Comment by fkantelberg** [comment]
Yes I haven't used format-patch again and the commits were shared between the vault and vault_share PR. Was weird and I re-did the entire process.

File: vault_share/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# © 2021-2024 Florian Kantelberg - initOS GmbH
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Vault - Share",
+    "summary": "Implementation of a mechanism to share secrets",
+    "license": "AGPL-3",
+    "version": "16.0.1.1.1",
```
Comment (pedrobaeza) [inline vault_share/__manifest__.py]:
```suggestion
    "version": "16.0.1.0.0",
```

File: vault_share/migrations/16.0.1.1.1/post-migrate.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# © 2024 Florian Kantelberg - initOS GmbH
```
Comment (pedrobaeza) [inline vault_share/migrations/16.0.1.1.1/post-migrate.py]: Change the folder name

File: vault_share/models/res_config_settings.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    vault_share_delay = fields.Integer(
+        string="Delayed Deletion",
+        related="company_id.vault_share_delay",
+        readonly=False,
+        help="Delays the deletion of a share. After the expiration date it continues "
+        "to stay inaccessible",
+    )
+
+    @api.onchange("vault_share_delay")
+    def _on_change_mins(self):
```
Comment (pedrobaeza) [inline vault_share/models/res_config_settings.py]: Follow naming guidelines:

```suggestion
    def _on_change_vault_share_delay(self):
```

File: vault_share/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+This module implements possibilities to share specific secrets with external users. This bases on the vault implementation and the generated RSA key pair.
+
+Share
+=====
```
Comment (pedrobaeza) [inline vault_share/readme/DESCRIPTION.rst]: Incompatible marking when the full README is composed.

```suggestion
~~~~~
```

File: vault_share/static/description/icon.png
Comment (pedrobaeza) [inline vault_share/static/description/icon.png:1]: New standard size is 140x140. Don't you have the SVG?

Comment (fkantelberg) [reply vault_share/static/description/icon.png:1]: There was never a svg and I'm not good enough with these tools. I just used https://spilymp.github.io/ibo/ again with the new size.

**Comment by pedrobaeza** [comment]
Let's not be more picky asking to squash this last commit, but there's no advantage in GitHub putting separate commits for new additions in a PR, and the commit history is a bit worse having them split.

/ocabot merge nobump

**Pull request merged**